src/util/saveArticles.ts

```diff
@@ -402,9 +403,33 @@ async function treatVideo(mw: MediaWiki, dump: Dump, srcCache: KVS<boolean>, art
     }
 
     sourceEl.setAttribute('src', newUrl);
+
+    /* Scrape subtitle */
+    const trackEle = videoEl.querySelector('track');
+    if (!trackEle) {
+        return { mediaDependencies };
+    }
+
+    const { content, title } = await treatSubtitles(trackEle, downloader, webUrlHost, mw);
+    const article = new ZimArticle({ url: `${title}.vtt`, mimeType: 'text/vtt', data: content, ns: 'I' });
+    zimCreator.addArticle(article);
+    trackEle.setAttribute('src', `${getRelativeFilePath(articleId, title, 'I')}.vtt`);
     return { mediaDependencies };
 }
 
+export async function treatSubtitles(trackEle: any, downloader: Downloader, webUrlHost: string, mw: MediaWiki): Promise<{ content: Buffer | string, title: string }> {
+    try {
+        const sourceUrl = getFullUrl(webUrlHost, trackEle.getAttribute('src'), mw.base);
+        const {trackformat, title} = QueryStringParser.parse(sourceUrl) as {title: string, trackformat: string};
+
+        const vttSourceUrl = sourceUrl.replace(trackformat, 'vtt');
```

Comment:
Super weak! what will happen if for some reason there is `srt` written somewhere else in the URL!